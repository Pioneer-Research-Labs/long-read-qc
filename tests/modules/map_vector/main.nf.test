nextflow_process {

    name "Test Process map_vector"
    script "modules/map_vector/main.nf"
    process "map_vector"

    test("Should map reads to vector"){

        when {
            params {
                outdir = outputDir
            }
            process {
                """
                input[0] = Channel.fromList([ tuple([ id:'Gateway_EmptyBarcodedLib1_20241217',
                genome:'H_elongata'],
                file("${projectDir}/test_data/Gateway_EmptyBarcodedLib1_20241217/5HKJT6_1_sample_1.fastq.gz"),
                file('${projectDir}/test_data/Gateway_EmptyBarcodedLib1_20241217/c.00323.dna'))])
                """
            }
        }

        then {
            assert process.success
            assert snapshot("$outputDir/Gateway_EmptyBarcodedLib1_20241217/mapped_vector_stats.tsv").match()
         }
    }

}
