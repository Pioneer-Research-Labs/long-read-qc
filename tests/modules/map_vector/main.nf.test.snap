{
    "Should map reads to vector": {
        "content": [
            "mapped_vector_stats.tsv:md5,cfecbeb9f8a363ab03f44ebcb0e1219d"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-15T13:26:43.420295"
    }
}